jwql/website/apps/jwql/data_containers.py

```diff
@@ -882,35 +899,189 @@ def get_thumbnails_all_instruments(instruments):
+                        if effexptm_min:
+                            if result['effexptm'] > int(effexptm_min):
+                                if filters:
+                                    if result['filter'] in filters[inst.lower()]:
+                                        filename = result['filename'].split('.')[0]
+                                        filenames.append(filename)
+                                else:
+                                    filename = result['filename'].split('.')[0]
+                                    filenames.append(filename)
+                        else:
+                            if filters:
+                                if result['filter'] in filters[inst.lower()]:
+                                    filename = result['filename'].split('.')[0]
+                                    filenames.append(filename)
+                            else:
+                                filename = result['filename'].split('.')[0]
+                                filenames.append(filename)
+                else:
+                    if effexptm_min:
+                        if result['effexptm'] > int(effexptm_min):
+                            if filters:
+                                if result['filter'] in filters[inst.lower()]:
+                                    filename = result['filename'].split('.')[0]
+                                    filenames.append(filename)
+                            else:
+                                filename = result['filename'].split('.')[0]
+                                filenames.append(filename)
+                    else:
+                        if filters:
+                            if result['filter'] in filters[inst.lower()]:
+                                filename = result['filename'].split('.')[0]
+                                filenames.append(filename)
+                        else:
+                            filename = result['filename'].split('.')[0]
+                            filenames.append(filename)
 
         # Get list of all thumbnails
         thumbnails = glob.glob(os.path.join(THUMBNAIL_FILESYSTEM, '*', '*.thumb'))
-
-    thumbnail_list.extend(thumbnails)
+        thumbnail_list.extend(thumbnails)
 
     # Get subset of preview images that match the filenames
-    thumbnails = [os.path.basename(item) for item in thumbnail_list if
-                  os.path.basename(item).split('_integ')[0] in filenames]
+    thumbnails_subset = [os.path.basename(item) for item in thumbnail_list if
+                         os.path.basename(item).split('_integ')[0] in filenames]
 
-    return thumbnails
+    # Eliminate any duplicates
+    thumbnails_subset = list(set(thumbnails_subset))
+
+    # Determine whether or not queried anomalies are flagged
+    final_subset = []
+    for thumbnail in thumbnails_subset:
+        rootname = thumbnail.split("_")[0]+"_"+thumbnail.split("_")[1]+"_"+thumbnail.split("_")[2]+"_"+thumbnail.split("_")[3]
```

Comment:
Might be better to use string fornatting here, i.e.

```python
components = thumbnail.split('_')
rootname = '{}_{}_{}_{}'.format(components[0], components[1], components[2], components[3])

Reply:
Good idea!